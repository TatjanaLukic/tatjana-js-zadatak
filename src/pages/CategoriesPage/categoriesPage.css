@import url("https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,400;0,500;0,700;1,400&display=swap");
.categories-page {
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  font-family: "Roboto", sans-serif;
  padding: 0.8rem;
}
@media screen and (min-width: 576px) {
  .categories-page {
    padding: 1rem;
  }
}
@media screen and (min-width: 992px) {
  .categories-page {
    width: 75%;
    margin: auto;
  }
}

.categories-page__title {
  font-size: 1.3rem;
  font-weight: bold;
  padding: 0 0.3rem;
  margin: 0.8rem 0;
}
@media screen and (min-width: 576px) {
  .categories-page__title {
    font-size: 1.5rem;
    font-weight: 600;
    padding: 0 2rem;
    margin: 1rem 0;
  }
}

.categories-page__blocks--wrapper {
  border: 2px solid #ddd;
  border-radius: 3%;
}

/*# sourceMappingURL=categoriesPage.css.map */
