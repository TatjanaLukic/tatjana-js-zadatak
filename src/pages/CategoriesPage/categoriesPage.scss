@import "../../assets/variables";
.categories-page{
    display: flex;
    flex-direction: column;
    box-sizing: border-box;
    font-family: 'Roboto', sans-serif;
    padding: 0.8rem;

  
    @media screen and (min-width: $smallSize) {
      padding: 1rem;
    }
  
    @media screen and (min-width: $largeSize) {
      width: 75%;
      margin: auto;
    }
  }

  .categories-page__title {
    font-size: 1.3rem;
    font-weight: bold;
    padding: 0 0.3rem;
    margin: 0.8rem 0;
  
    @media screen and (min-width: $smallSize) {
      font-size: 1.5rem;
      font-weight: 600;
      padding: 0 2rem;
      margin: 1rem 0;
    }
  }

  .categories-page__blocks--wrapper {
      border: 2px solid $categories-border;
      border-radius: 3%;
  }
  

