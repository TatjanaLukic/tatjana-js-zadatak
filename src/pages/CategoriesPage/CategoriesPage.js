import React, { useContext } from "react";
import CountryContext from "../../context/countryContext";
import CategoryBlock from "../../components/CategoryBlock/CategoryBlock";
import "./categoriesPage.css";

const CategoriesPage = () => {
  const country = useContext(CountryContext);
  const countryName =
    country.country === "us" ? "United States" : "Great Britain";

  const categories = [
    "entertainment",
    "general",
    "health",
    "science",
    "sport",
    "technology",
  ];
  return (
    <div className="categories-page">
      <h1 className="categories-page__title">
        Top 5 news by categories from {countryName}
      </h1>
      <div className="categories-page__blocks--wrapper">
        {categories.map((categoryName, inx) => {
          return <CategoryBlock key={inx} categoryName={categoryName} />;
        })}
      </div>
    </div>
  );
};

export default CategoriesPage;
