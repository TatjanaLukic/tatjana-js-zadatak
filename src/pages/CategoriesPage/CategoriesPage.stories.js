import React from "react";
import { MemoryRouter } from "react-router-dom";
import CategoriesPage from "./CategoriesPage";

export default {
  title: "App/CategoriesPage",
  component: CategoriesPage
};

const Template = () => <CategoriesPage />;

export const Default = Template.bind({});
Default.decorators=  [(getStory) => <MemoryRouter>{getStory()}</MemoryRouter>]


