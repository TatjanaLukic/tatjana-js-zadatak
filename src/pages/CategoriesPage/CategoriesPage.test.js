import React from "react";
import { shallow } from "enzyme";
import CategoriesPage from "./CategoriesPage";

describe("<CategoriesPage />", () => {
  let categoriesPage;

  beforeEach(() => {
    categoriesPage = shallow(<CategoriesPage />);
  });

  it("should have h1", () => {
    expect(categoriesPage.find("h1")).toHaveLength(1);
  });
  it("should have title=Top 5 news by categories from United States", () => {
    expect(categoriesPage.find("h1").text()).toEqual("Top 5 news by categories from United States")
  });
  it("should have 5 CategoryBlock", () => {
    expect(categoriesPage.find("CategoryBlock")).toHaveLength(6);
  });
  it("spanshot matches", () => {
    expect(categoriesPage).toMatchSnapshot();
  });
});
