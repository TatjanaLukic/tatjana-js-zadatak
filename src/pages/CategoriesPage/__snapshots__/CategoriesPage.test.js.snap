// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoriesPage /> spanshot matches 1`] = `
<div
  className="categories-page"
>
  <h1
    className="categories-page__title"
  >
    Top 5 news by categories from 
    United States
  </h1>
  <div
    className="categories-page__blocks--wrapper"
  >
    <CategoryBlock
      categoryName="entertainment"
      key="0"
    />
    <CategoryBlock
      categoryName="general"
      key="1"
    />
    <CategoryBlock
      categoryName="health"
      key="2"
    />
    <CategoryBlock
      categoryName="science"
      key="3"
    />
    <CategoryBlock
      categoryName="sport"
      key="4"
    />
    <CategoryBlock
      categoryName="technology"
      key="5"
    />
  </div>
</div>
`;
